Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -65,10 +68,19 @@ - (id)initWithDrawingAreaProxy:(WebKit::RemoteLayerTreeDrawingAreaProxy*)drawing
     if (self = [super init]) {
         _drawingAreaProxy = drawingAreaProxy;
         // Note that CADisplayLink retains its target (self), so a call to -invalidate is needed on teardown.
+#if ENABLE(CUSTOM_DISPLAY_UPDATE_RATES)
+        NSInteger framerateOverride = [NSUserDefaults.standardUserDefaults integerForKey:@"MainScreenRefreshRate"];
+        _displayLink = [CADisplayLink displayLinkWithTarget:self selector:framerateOverride ? @selector(emptyDisplayLinkFired:) : @selector(displayLinkFired:)];
+#else
         _displayLink = [CADisplayLink displayLinkWithTarget:self selector:@selector(displayLinkFired:)];
+#endif
         [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
         _displayLink.paused = YES;
         _displayLink.preferredFramesPerSecond = 60;
+#if ENABLE(CUSTOM_DISPLAY_UPDATE_RATES)
+        if (framerateOverride)
+            _updateTimer = [NSTimer scheduledTimerWithTimeInterval:1.0 / framerateOverride target:self selector:@selector(timerFired) userInfo:nil repeats:YES];
```

Comment:
Looks like both _displayLink and _updateTimer aren't retained. Are we just relying on the fact that they keep a ref count while active?

Reply:
I'm not sure about the display link but looks like _updateTimer is retained from the recurring nature of the timer. Probably would be more clear if I retained it.

Reply:
I added retain / release to the updateTimer @smfr but left the displayLink as-is since we don't retain it in the `!ENABLE(CUSTOM_DISPLAY_UPDATE_RATES)` path

Reply:
The runloop retains the display link while it's installed. Is a bit sketch that we don't keep a reference too, but it's fine.